dev/extra-ml-requirements.txt

```diff
@@ -4,35 +4,35 @@
 # Required by mlflow.azureml
 azureml-sdk==1.2.0; python_version >= "3.0"
 # Required by mlflow.pyfunc
-cloudpickle==0.8.0
+cloudpickle
```

Comment:
This isn't actually an extra ML requirement (I should've caught this while reviewing Sid's PR). It's included in MLflow by default via `setup.py`. We should remove it from this file.